tstest/integration/integration_test.go

```diff
@@ -1412,14 +1413,32 @@ func TestLogoutRemovesAllPeers(t *testing.T) {
 	wantNode0PeerCount(expectedPeers) // all existing peers and the new node
 }
 
-func TestAutoUpdateDefaults(t *testing.T) {
-	if !feature.CanAutoUpdate() {
-		t.Skip("auto-updates not supported on this platform")
+func TestAutoUpdateDefaults(t *testing.T)     { testAutoUpdateDefaults(t, false) }
+func TestAutoUpdateDefaults_cap(t *testing.T) { testAutoUpdateDefaults(t, true) }
+
+// useCap is whether to use NodeAttrDefaultAutoUpdate (as opposed to the old
+// DeprecatedDefaultAutoUpdate top-level MapResponse field).
+func testAutoUpdateDefaults(t *testing.T, useCap bool) {
+	if was, ok := feature.HookCanAutoUpdate.GetOk(); ok {
+		defer func() { feature.HookCanAutoUpdate.SetForTest(was) }()
 	}
+	feature.HookCanAutoUpdate.SetForTest(func() bool { return true })
+
 	tstest.Shard(t)
-	tstest.Parallel(t)
 	env := NewTestEnv(t)
 
+	var (
+		modifyMu               sync.Mutex
+		modifyFirstMapResponse func(mr *tailcfg.MapResponse, req *tailcfg.MapRequest)
```

Comment:
(optional)
```suggestion
		modifyFirstMapResponse = func(mr *tailcfg.MapResponse, req *tailcfg.MapRequest) {}
```

so that we don't need to check the condition below?